internal/hmyapi/apiv2/transactionpool.go

```diff
@@ -120,55 +124,107 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
+	if block, _ := s.b.BlockByNumber(ctx, blockNr); block != nil {
+		return newRPCStakingTransactionFromBlockIndex(block, uint64(index))
 	}
 	return nil
 }
 
 // GetStakingTransactionByBlockHashAndIndex returns the transaction for the given block hash and index.
-func (s *PublicTransactionPoolAPI) GetStakingTransactionByBlockHashAndIndex(ctx context.Context, blockHash common.Hash, index uint64) *RPCStakingTransaction {
+func (s *PublicTransactionPoolAPI) GetStakingTransactionByBlockHashAndIndex(ctx context.Context, blockHash common.Hash, index hexutil.Uint) *RPCStakingTransaction {
 	if block, _ := s.b.GetBlock(ctx, blockHash); block != nil {
-		return newRPCStakingTransactionFromBlockIndex(block, index)
+		return newRPCStakingTransactionFromBlockIndex(block, uint64(index))
+	}
+	return nil
+}
+
+// GetStakingTransactionByHash returns the staking transaction for the given hash
+func (s *PublicTransactionPoolAPI) GetStakingTransactionByHash(ctx context.Context, hash common.Hash) *RPCStakingTransaction {
+	// Try to return an already finalized transaction
+	stx, blockHash, blockNumber, index := rawdb.ReadStakingTransaction(s.b.ChainDb(), hash)
+	block, _ := s.b.GetBlock(ctx, blockHash)
+	if block == nil {
+		return nil
+	}
+	if stx != nil {
+		return newRPCStakingTransaction(stx, blockHash, blockNumber, block.Time().Uint64(), index)
 	}
+	// Transaction unknown, return as such
 	return nil
 }
 
-// GetTransactionCount returns the number of transactions the given address has sent for the given block number
-func (s *PublicTransactionPoolAPI) GetTransactionCount(ctx context.Context, addr string, blockNr uint64) (uint64, error) {
+// GetTransactionCount returns the number of transactions the given address has sent from genesis to the input block number
+// NOTE: unlike other txn apis where staking vs. regular txns are separate,
+// the transaction count here includes the count of both regular and staking txns
+func (s *PublicTransactionPoolAPI) GetTransactionCount(ctx context.Context, addr string, blockNr rpc.BlockNumber) (*hexutil.Uint64, error) {
 	address := internal_common.ParseAddr(addr)
 	// Ask transaction pool for the nonce which includes pending transactions
-	if rpc.BlockNumber(blockNr) == rpc.PendingBlockNumber {
+	if blockNr == rpc.PendingBlockNumber {
 		nonce, err := s.b.GetPoolNonce(ctx, address)
 		if err != nil {
-			return 0, err
+			return nil, err
 		}
-		return nonce, nil
+		return (*hexutil.Uint64)(&nonce), nil
 	}
 	// Resolve block number and use its state to ask for the nonce
-	state, _, err := s.b.StateAndHeaderByNumber(ctx, rpc.BlockNumber(blockNr))
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, blockNr)
 	if state == nil || err != nil {
-		return 0, err
+		return nil, err
 	}
 	nonce := state.GetNonce(address)
-	return nonce, state.Error()
+	return (*hexutil.Uint64)(&nonce), state.Error()
```

Comment:
same comment here. looks like you are using uint64. please use the same throughout.

Reply:
ditto. Not my change. will revert.